libmamba/src/specs/regex_spec.cpp

```diff
@@ -39,25 +38,48 @@ namespace mamba::specs
     }
 
     RegexSpec::RegexSpec()
-        : RegexSpec(std::regex(free_pattern.data(), free_pattern.size()), std::string(free_pattern))
+        : RegexSpec(std::string(free_pattern))
     {
     }
 
-    RegexSpec::RegexSpec(std::regex pattern, std::string raw_pattern)
-        : m_pattern(std::move(pattern))
-        , m_raw_pattern(std::move(raw_pattern))
+    RegexSpec::RegexSpec(std::string raw_pattern)
     {
+        // Construct ss from raw_pattern, in particular make sure to replace all `*` by `.*`
+        // in the pattern if they are not preceded by a `.`.
```

Comment:
Wouldn't this mean that a glob string `py3.*` would match `py310` when it shouldn't have? i.e. we should first escape the existing `.` to `\.`, and then replace `*` by `.*`?

Reply:
Arf, I support your remark, but this clashes with behavior:

https://github.com/mamba-org/mamba/blob/374228f860033229be042c80f2515089b77f1c5b/libmamba/tests/src/specs/test_chimera_string_spec.cpp#L43-L57

Reply:
In `conda`, the "full regex mode" only kicks in if the build string is wrapped in `^...$`. Otherwise, it's understood as a normal glob string. See [docstring](https://github.com/conda/conda/blob/52b6393d6331e8aa36b2e23ab65766a980f381d2/conda/models/match_spec.py#L134-L139 and [implementation](https://github.com/conda/conda/blob/52b6393d6331e8aa36b2e23ab65766a980f381d2/conda/models/match_spec.py#L889-L894).